### ISSUE TITLE
[ROCM] llama2-7b-int4-rocm compilation failure

### ISSUE THREAD

**Issue by PhaneeshB** [iree-org/iree#14920]
### What happened?

llama2-7b-int4 compilation for rocm backend errs out with 
`error: 'func.func' op uses 1114112 bytes of shared memory; exceeded the limit of 166912 bytes`

Full diagnostics:
```
Diagnostics:
<stdin>:1727:11: error: 'func.func' op uses 1114112 bytes of shared memory; exceeded the limit of 166912 bytes
    %43 = linalg.generic {indexing_maps = [#map15_0, #map16_0, #map17_0], iterator_types = ["parallel", "parallel", "parallel", "reduction", "reduction"]} ins(%expanded_751, %42 : tensor<1x?x32x128xf16>, tensor<4096x32x128xf16>) outs(%41 : tensor<1x?x4096xf16>) {
          ^
<stdin>:809:3: note: called from
  func.func @first_vicuna_forward(%arg0: tensor<1x?xi64>) -> (tensor<1x?x32000xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>) {
  ^
<stdin>:1727:11: error: failed to run translation of source executable to target executable for backend #hal.executable.target<"rocm", "rocm-hsaco-fb", {target_arch = "gfx1100"}>
    %43 = linalg.generic {indexing_maps = [#map15_0, #map16_0, #map17_0], iterator_types = ["parallel", "parallel", "parallel", "reduction", "reduction"]} ins(%expanded_751, %42 : tensor<1x?x32x128xf16>, tensor<4096x32x128xf16>) outs(%41 : tensor<1x?x4096xf16>) {
          ^
<stdin>:809:3: note: called from
  func.func @first_vicuna_forward(%arg0: tensor<1x?xi64>) -> (tensor<1x?x32000xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>) {
  ^
<stdin>:1727:11: error: failed to serialize executables
    %43 = linalg.generic {indexing_maps = [#map15_0, #map16_0, #map17_0], iterator_types = ["parallel", "parallel", "parallel", "reduction", "reduction"]} ins(%expanded_751, %42 : tensor<1x?x32x128xf16>, tensor<4096x32x128xf16>) outs(%41 : tensor<1x?x4096xf16>) {
          ^
<stdin>:809:3: note: called from
  func.func @first_vicuna_forward(%arg0: tensor<1x?xi64>) -> (tensor<1x?x32000xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>) {
  ^
<stdin>:2035:12: error: 'func.func' op uses 1114112 bytes of shared memory; exceeded the limit of 166912 bytes
    %111 = linalg.generic {indexing_maps = [#map15_0, #map16_0, #map17_0], iterator_types = ["parallel", "parallel", "parallel", "reduction", "reduction"]} ins(%expanded_781, %110 : tensor<1x?x32x128xf16>, tensor<11008x32x128xf16>) outs(%109 : tensor<1x?x11008xf16>) {
           ^
<stdin>:809:3: note: called from
  func.func @first_vicuna_forward(%arg0: tensor<1x?xi64>) -> (tensor<1x?x32000xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>) {
  ^
<stdin>:2035:12: error: failed to run translation of source executable to target executable for backend #hal.executable.target<"rocm", "rocm-hsaco-fb", {target_arch = "gfx1100"}>
    %111 = linalg.generic {indexing_maps = [#map15_0, #map16_0, #map17_0], iterator_types = ["parallel", "parallel", "parallel", "reduction", "reduction"]} ins(%expanded_781, %110 : tensor<1x?x32x128xf16>, tensor<11008x32x128xf16>) outs(%109 : tensor<1x?x11008xf16>) {
           ^
<stdin>:809:3: note: called from
  func.func @first_vicuna_forward(%arg0: tensor<1x?xi64>) -> (tensor<1x?x32000xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>) {
  ^
<stdin>:2035:12: error: failed to serialize executables
    %111 = linalg.generic {indexing_maps = [#map15_0, #map16_0, #map17_0], iterator_types = ["parallel", "parallel", "parallel", "reduction", "reduction"]} ins(%expanded_781, %110 : tensor<1x?x32x128xf16>, tensor<11008x32x128xf16>) outs(%109 : tensor<1x?x11008xf16>) {
           ^
<stdin>:809:3: note: called from
  func.func @first_vicuna_forward(%arg0: tensor<1x?xi64>) -> (tensor<1x?x32000xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>) {
  ^
<stdin>:2093:12: error: 'func.func' op uses 2994176 bytes of shared memory; exceeded the limit of 166912 bytes
    %123 = linalg.generic {indexing_maps = [#map15_0, #map16_0, #map17_0], iterator_types = ["parallel", "parallel", "parallel", "reduction", "reduction"]} ins(%expanded_787, %122 : tensor<1x?x86x128xf16>, tensor<4096x86x128xf16>) outs(%121 : tensor<1x?x4096xf16>) {
           ^
<stdin>:809:3: note: called from
  func.func @first_vicuna_forward(%arg0: tensor<1x?xi64>) -> (tensor<1x?x32000xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>) {
  ^
<stdin>:2093:12: error: failed to run translation of source executable to target executable for backend #hal.executable.target<"rocm", "rocm-hsaco-fb", {target_arch = "gfx1100"}>
    %123 = linalg.generic {indexing_maps = [#map15_0, #map16_0, #map17_0], iterator_types = ["parallel", "parallel", "parallel", "reduction", "reduction"]} ins(%expanded_787, %122 : tensor<1x?x86x128xf16>, tensor<4096x86x128xf16>) outs(%121 : tensor<1x?x4096xf16>) {
           ^
<stdin>:809:3: note: called from
  func.func @first_vicuna_forward(%arg0: tensor<1x?xi64>) -> (tensor<1x?x32000xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>) {
  ^
<stdin>:2093:12: error: failed to serialize executables
    %123 = linalg.generic {indexing_maps = [#map15_0, #map16_0, #map17_0], iterator_types = ["parallel", "parallel", "parallel", "reduction", "reduction"]} ins(%expanded_787, %122 : tensor<1x?x86x128xf16>, tensor<4096x86x128xf16>) outs(%121 : tensor<1x?x4096xf16>) {
           ^
<stdin>:809:3: note: called from
  func.func @first_vicuna_forward(%arg0: tensor<1x?xi64>) -> (tensor<1x?x32000xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>, tensor<1x32x?x128xf16>) {
  ^
```

### Steps to reproduce your issue

mlir file : https://storage.googleapis.com/shark_tank/llama2_7b/unsharded/mlir/llama2_7b_int4.mlir

iree-compile command :
```
 iree-compile --iree-input-type=tm_tensor --iree-vm-bytecode-module-output-format=flatbuffer-binary --iree-hal-target-backends=rocm --mlir-print-debuginfo --mlir-print-op-on-diagnostic=false --iree-llvmcpu-target-cpu-features=host --iree-rocm-target-chip=gfx1100 --iree-rocm-link-bc=true --iree-rocm-bc-dir=/opt/rocm/amdgcn/bitcode --iree-stream-resource-max-allocation-size=4294967295 --iree-vm-bytecode-module-strip-source-map=true --iree-util-zero-fill-elided-attrs --iree-opt-strip-assertions=true --verify=false --iree-vm-target-truncate-unsupported-floats --iree-codegen-check-ir-before-llvm-conversion=false --iree-vm-bytecode-module-output-format=flatbuffer-binary <input.mlir> -o <output.vmfb>
```

iree commit :  de4f4e539a

**Comment by PhaneeshB** [comment]
On looking at it further, found that the limit of `166912` comes from [Codegen/LLVMGPU/Passes.cpp](https://github.com/openxla/iree/blob/de4f4e539a1e1dd7f0ffe2ec2eee32b4587faa12/compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp#L526)

**Comment by raikonenfnu** [comment]
So a workaround is to disable the fusion of dequant-matvec for dynamic shapes. But ideally, we just fix it correctly. :)

**Comment by Abhishek-Varma** [comment]
For CUDA too we're facing this issue for both 7B and 13B llama2, where the invoke command is :-
```
iree-compile --iree-input-type=tm_tensor --iree-vm-bytecode-module-output-format=flatbuffer-binary --iree-hal-target-backends=cuda --mlir-print-debuginfo --mlir-print-op-on-diagnostic=false --iree-llvmcpu-target-cpu-features=host --iree-stream-resource-max-allocation-size=4294967295 --iree-vm-bytecode-module-strip-source-map=true --iree-util-zero-fill-elided-attrs --iree-opt-strip-assertions=true --verify=false --iree-vm-target-truncate-unsupported-floats --iree-codegen-check-ir-before-llvm-conversion=false --iree-vm-bytecode-module-output-format=flatbuffer-binary
```

**Comment by qedawkins** [comment]
This should be fixed by #14922.
